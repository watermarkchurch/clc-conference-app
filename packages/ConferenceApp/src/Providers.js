import React from 'react';
import { Providers } from '@apollosproject/ui-kit';
import ClientProvider from './client';
import theme from './theme';

const AppProvider = (props) => (
  <Providers themeInput={theme}>
    <ClientProvider {...props} />
  </Providers>
);

export default AppProvider;
