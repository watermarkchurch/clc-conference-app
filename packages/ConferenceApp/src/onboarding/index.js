import React, { Component } from 'react';
import {
  View,
  Animated,
  StyleSheet,
  Image,
  StatusBar,
  Platform,
} from 'react-native';
import { withApollo } from 'react-apollo';
import SafeAreaView from 'react-native-safe-area-view';
import { compose } from 'recompose';
import OneSignal from 'react-native-onesignal';
import PropTypes from 'prop-types';
import gql from 'graphql-tag';

import {
  H2,
  BodyText,
  styled,
  withTheme,
  withThemeMixin,
  Button,
} from '@apollosproject/ui-kit';

import Swiper from 'react-native-swiper';

export Prompt from './Prompt';

const style = StyleSheet.create({
  slideImage: {
    flex: 1,
    resizeMode: 'cover',
  },
});

const getOnboarded = gql`
  query {
    didOnboard @client
  }
`;

const Slide = compose(
  withThemeMixin(({ themeMixin }) => themeMixin),
  styled(({ theme }) => ({
    flex: 1,
    justifyContent: 'center',
    paddingLeft: theme.sizing.baseUnit * 2,
    paddingRight: theme.sizing.baseUnit * 3,
  }))
)(View);

const ThemedSwiper = withTheme(() => ({
  dotColor: 'rgba(0,0,0,0.25)',
  activeDotColor: 'rgba(0,0,0,0.75)',
}))(Swiper);

const SlideTitle = styled(({ theme }) => ({
  paddingBottom: theme.sizing.baseUnit,
}))(H2);

const PaddedView = styled(({ theme }) => ({
  paddingHorizontal: theme.sizing.baseUnit * 2,
  paddingVertical: theme.sizing.baseUnit / 2,
}))(View);

const LogoWrapper = styled({
  position: 'absolute',
  top: 0,
  left: 0,
})(View);

const LogoImage = styled(({ theme }) => ({
  height: theme.sizing.baseUnit * 3,
  aspectRatio: 1,
  margin: theme.sizing.baseUnit * 2,
}))(Image);

const ActionRow = styled({
  position: 'absolute',
  bottom: 0,
  left: 0,
  right: 0,
})(SafeAreaView);

class Onboarding extends Component {
  static navigationOptions = {
    header: null,
  };

  static propTypes = {
    client: PropTypes.shape({
      writeQuery: PropTypes.func,
    }),
  };

  scrollPos = new Animated.Value(0);

  viewWidth = new Animated.Value(1);

  componentDidMount() {
    this.props.client.writeQuery({
      query: getOnboarded,
      data: {
        didOnboard: true,
      },
    });
  }

  handlePromptNotifications = () => {
    OneSignal.registerForPushNotifications();
    this.handleClose();
  };

  handleClose = () => {
    this.props.navigation.goBack();
  };

  render() {
    const animatedIndex = Animated.divide(this.scrollPos, this.viewWidth);
    return (
      <View
        style={StyleSheet.absoluteFill}
        onLayout={Animated.event([
          {
            nativeEvent: { layout: { width: this.viewWidth } },
          },
        ])}
      >
        <StatusBar hidden />
        <View style={StyleSheet.absoluteFill}>
          <Animated.Image
            source={require('./slide-orange.jpg')}
            style={style.slideImage}
          />
        </View>
        <View style={StyleSheet.absoluteFill}>
          <Animated.Image
            source={require('./slide-purple.jpg')}
            style={[style.slideImage, { opacity: animatedIndex }]}
          />
        </View>
        <View style={StyleSheet.absoluteFill}>
          <Animated.Image
            source={require('./slide-gray.jpg')}
            style={[
              style.slideImage,
              { opacity: Animated.subtract(animatedIndex, 1) },
            ]}
          />
        </View>
        <LogoWrapper>
          <SafeAreaView forceInset={{ top: 'always' }}>
            <LogoImage source={require('./logo.png')} />
          </SafeAreaView>
        </LogoWrapper>
        <View style={StyleSheet.absoluteFill}>
          <ThemedSwiper
            loop={false}
            showsButtons={false}
            onScroll={Animated.event([
              { nativeEvent: { contentOffset: { x: this.scrollPos } } },
            ])}
            scrollEventThrottle={16}
          >
            <Slide themeMixin={{ type: 'dark' }}>
              <SlideTitle>Meet the Church Leaders Conference App</SlideTitle>
              <BodyText>
                Your go-to resource for schedule, maps, news and of course -
                Farkle rules.
              </BodyText>
            </Slide>
            <Slide themeMixin={{ type: 'dark' }}>
              <SlideTitle>Customize Your Schedule</SlideTitle>
              <BodyText>
                Explore the schedule and tracks based on specific ministry
                areas. Star your favorites and they will be added to your{' '}
                {'"My CLC"'} tab.
              </BodyText>
            </Slide>
            {Platform.select({
              ios: (
                <Slide>
                  <SlideTitle>{"Don't Miss a Thing"}</SlideTitle>
                  <BodyText>
                    Allow push notifications to stay up to date on last minute
                    announcements, room changes and Farkle challenges.
                  </BodyText>
                  <ActionRow forceInset={{ bottom: 'always' }}>
                    <PaddedView>
                      <Button
                        title="Enable Notifications"
                        onPress={this.handlePromptNotifications}
                      />
                    </PaddedView>
                    <PaddedView vertical={false}>
                      <Button
                        title="No thanks"
                        type="default"
                        onPress={this.handleClose}
                      />
                    </PaddedView>
                    <PaddedView />
                  </ActionRow>
                </Slide>
              ),
              android: (
                <Slide>
                  <SlideTitle>{"Don't Miss a Thing"}</SlideTitle>
                  <BodyText>
                    {
                      "You'll receive push notifications to stay up to date on last minute announcements, room changes and Farkle challenges."
                    }
                  </BodyText>
                  <ActionRow forceInset={{ bottom: 'always' }}>
                    <PaddedView>
                      <Button title="Let's Go!" onPress={this.handleClose} />
                    </PaddedView>
                    <PaddedView />
                    <PaddedView />
                  </ActionRow>
                </Slide>
              ),
            })}
          </ThemedSwiper>
        </View>
      </View>
    );
  }
}

export default withApollo(Onboarding);
