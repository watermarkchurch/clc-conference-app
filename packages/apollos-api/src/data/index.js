import { gql } from "apollo-server";

import { createApolloServerConfig } from "@apollosproject/server-core";

import * as Analytics from "@apollosproject/data-connector-analytics";
// import * as Scripture from "@apollosproject/data-connector-bible";
// import * as LiveStream from "@apollosproject/data-connector-church-online";
import * as Cloudinary from "@apollosproject/data-connector-cloudinary";
import * as OneSignal from "@apollosproject/data-connector-onesignal";
// import * as Pass from "@apollosproject/data-connector-passes";
// import {
//   Followings,
//   Interactions,
//   RockConstants,
//   Family,
//   Person,
//   ContentItem,
//   ContentChannel,
//   Sharable,
//   Auth,
//   PersonalDevice,
//   Template
// } from "@apollosproject/data-connector-rock";

import { peopleSchema } from "@apollosproject/data-schema";

const data = {
  Cloudinary,
  Analytics,
  OneSignal,
  People: { schema: peopleSchema }
};

const {
  dataSources,
  resolvers,
  schema,
  context,
  applyServerMiddleware
} = createApolloServerConfig(data);

export { dataSources, resolvers, schema, context, applyServerMiddleware };

// the upload Scalar is added
export const testSchema = [
  gql`
    scalar Upload
  `,
  ...schema
];
