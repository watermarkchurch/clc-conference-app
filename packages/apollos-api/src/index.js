import dotenv from 'dotenv/config'; // eslint-disable-line
import appdynamics from 'appdynamics';
import config from './config'; // eslint-disable-line
import server from './server';

export { testSchema } from './server'; // eslint-disable-line import/prefer-default-export

if (
  process.env.APPD_HOST &&
  process.env.APPD_ACCOUNTNAME &&
  process.env.APPD_ACCESSKEY
) {
  appdynamics.profile({
    controllerHostName: process.env.APPD_HOST,
    controllerPort: 443,
    controllerSslEnabled: true,
    accountName: process.env.APPD_ACCOUNTNAME,
    accountAccessKey: process.env.APPD_ACCESSKEY,
    applicationName: 'wcc-clc-api',
    tierName: 'heroku-web',
    nodeName: 'heroku-dyno',
  });
}

// Use the port, if provided.
const { PORT } = process.env;
if (!PORT && process.env.NODE_ENV !== 'test')
  console.warn(
    'Add `ENV=4000` if you are having trouble connecting to the server. By default, PORT is random.'
  );

server.listen({ port: PORT }, () => {
  console.log(`ğŸš€ Server ready at http://0.0.0.0:${PORT}`);
});
