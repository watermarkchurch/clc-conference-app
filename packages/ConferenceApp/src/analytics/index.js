/* eslint-disable */
import gql from 'graphql-tag';
import DeviceInfo from 'react-native-device-info';
import { Platform } from 'react-native';
import Analytics from 'appcenter-analytics';

import { client } from '../client';

const anonymousId = DeviceInfo.getUniqueID();

const deviceInfo = {
  platform: Platform.OS === 'ios' ? 'iOS' : 'Android',
  deviceId: anonymousId,
  deviceModel: DeviceInfo.getModel(),
  osVersion: DeviceInfo.getSystemVersion(),
  appVersion: DeviceInfo.getVersion(),
};

const trackMutation = gql`
  mutation track($input: AnalyticsTrackInput!) {
    trackEvent(input: $input) {
      success
    }
  }
`;

const identifyMutation = gql`
  mutation identify($input: AnalyticsIdentifyInput!) {
    identifySelf(input: $input) {
      success
    }
  }
`;

const propertiesToGqlInput = (props = []) =>
  Object.keys(props).map(key => ({
    field: key,
    value: props[key],
  }));

export const track = ({ eventName, properties }) => {
  Analytics.trackEvent(eventName, properties);
};

export const identify = () => {};

export const events = {
  LikeContent: 'Like Content',
  UnlikeContent: 'Unlike Content',
  ViewContent: 'View Content',
  ShareContent: 'Share Content',
  UserLogin: 'User Login',
  UserSignup: 'User Signup',
  UserLogout: 'UserLogout',
  UserForgotPassword: 'User Forgot Password',
  UserPlayedMedia: 'User Played Media',
};

export default {
  track,
  identify,
  events,
};
