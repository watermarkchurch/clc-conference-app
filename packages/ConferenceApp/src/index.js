import React from 'react';
import { StatusBar } from 'react-native';
import { createStackNavigator, createAppContainer } from 'react-navigation';

import { BackgroundView, withTheme } from '@apollosproject/ui-kit';

import MediaPlayer from './ui/MediaPlayer';
import Providers from './Providers';
import { NotificationsManager } from './notifications';
import NavigationService from './NavigationService';
import ContentSingle from './content-single';
import Tabs from './tabs';
import Onboarding, { Prompt as OnboardingPrompt } from './onboarding';
import { track } from './analytics';

const AppStatusBar = withTheme(({ theme }) => ({
  barStyle: 'dark-content',
  backgroundColor: theme.colors.paper,
}))(StatusBar);

const AppNavigator = createAppContainer(
  createStackNavigator(
    {
      Tabs,
      ContentSingle,
      Onboarding,
    },
    {
      initialRouteName: 'Tabs',
      mode: 'modal',
      headerMode: 'screen',
    }
  )
);

function getActiveRoute(navigationState) {
  if (!navigationState) {
    return null;
  }
  const route = navigationState.routes[navigationState.index];
  // dive into nested navigators
  if (route.routes) {
    return getActiveRoute(route);
  }
  return route;
}

const App = () => (
  <Providers>
    <BackgroundView>
      <AppStatusBar barStyle="dark-content" />
      <AppNavigator
        ref={(navigatorRef) => {
          NavigationService.setTopLevelNavigator(navigatorRef);
        }}
        onNavigationStateChange={(prevState, currentState) => {
          const route = getActiveRoute(currentState);
          track({
            eventName: 'ScreenView',
            properties: {
              route: route.name,
              ...route.params,
            },
          });
        }}
      />
      <OnboardingPrompt />
      <NotificationsManager />
      <MediaPlayer />
    </BackgroundView>
  </Providers>
);

export default App;
