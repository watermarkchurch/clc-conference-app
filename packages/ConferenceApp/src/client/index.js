import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
// import SplashScreen from 'react-native-splash-screen';
import gql from 'graphql-tag';

import { resolvers, schema } from '../store';
import httpLink from './httpLink';
import cache, { ensureCacheHydration } from './cache';

const link = httpLink;

export const client = new ApolloClient({
  link,
  cache,
  queryDeduplication: true,
  shouldBatch: true,
  resolvers,
  typeDefs: schema,
});

export const CACHE_LOADED = gql`
  query {
    cacheLoaded @client
  }
`;

class ClientProvider extends PureComponent {
  static propTypes = {
    client: PropTypes.shape({
      cache: PropTypes.shape({}),
      mutate: PropTypes.func,
      writeQuery: PropTypes.func,
    }),
  };

  static defaultProps = {
    client,
  };

  async componentDidMount() {
    try {
      await ensureCacheHydration;
    } catch (e) {
      throw e;
    } finally {
      this.props.client.writeQuery({
        query: CACHE_LOADED,
        data: { cacheLoaded: true },
      });
      // if (SplashScreen && SplashScreen.hide) SplashScreen.hide();
    }
  }

  render() {
    return <ApolloProvider {...this.props} />;
  }
}

export default ClientProvider;
