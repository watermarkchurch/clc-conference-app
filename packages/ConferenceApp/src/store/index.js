import { merge, get } from 'lodash';
import gql from 'graphql-tag';
import { track, events } from '../analytics';
import { CACHE_LOADED } from '../client'; // eslint-disable-line
import updatePushId from '../notifications/updatePushId';
// TODO: this will require more organization...ie...not keeping everything in one file.
// But this is simple while our needs our small.

export const schema = `
  type Query {
    mediaPlayer: MediaPlayerState
    devicePushId: String
    cacheLoaded: Boolean
    likedContent: [Node]
  }

  extend type Node {
    isLiked: Boolean
  }

  type Mutation {
    mediaPlayerUpdateState(isPlaying: Boolean, isFullscreen: Boolean, isVisible: Boolean): Boolean
    mediaPlayerSetPlayhead(currentTime: Float): Boolean
    mediaPlayerPlayNow(
      parentId: ID,
      mediaSource: VideoMediaSource!,
      posterSources: [ImageMediaSource],
      title: String,
      artist: String,
      isVideo: Boolean,
    ): Boolean

    cacheMarkLoaded
    updateDevicePushId(pushId: String!)

    updateLikeEntity(input: LikeEntityInput!)
  }

  enum LIKE_OPERATION {
    Like
    Unlike
  }

  input LikeEntityInput {
    nodeId: ID!
    operation: LIKE_OPERATION!
  }

  type MediaPlayerState {
    currentTrack: MediaPlayerTrack
    isPlaying: Boolean
    isFullscreen: Boolean
    isVisible: Boolean
    currentTime: Float
  }

  type MediaPlayerProgress {
    currentTime: Float
    playableDuration: Float
    seekableDuration: Float
    duration: Float
  }

  type MediaPlayerTrack {
    id: ID!
    parentId: ID
    mediaSource: VideoMediaSource!
    posterSources: [ImageMediaSource]
    title: String
    artist: String
    isVideo: Boolean
  }
`;

export const defaults = {
  __typename: 'Query',
  cacheLoaded: false,
  pushId: null,
  mediaPlayer: {
    __typename: 'MediaPlayerState',
    currentTrack: null,
    isPlaying: false,
    isFullscreen: false,
    isVisible: false,
    currentTime: 0,
    showVideo: true,
    muted: false,
  },
  likedContent: [],
};

let trackId = 0;

const getIsLoggedIn = gql`
  query {
    isLoggedIn @client
  }
`;

export const resolvers = {
  Query: {
    node: (root, { id }, { cache }) => {
      const query = gql`
        query {
          likedContent @client {
            id
          }
        }
      `;

      let likedContent = [];

      try {
        ({ likedContent } = cache.readQuery({ query }));
      } catch (e) {
        likedContent = [];
      }

      return { isLiked: !!likedContent.find((content) => content.id === id) };
    },
  },
  Mutation: {
    updateLikeEntity: (root, { input }, { cache }) => {
      const query = gql`
        query {
          likedContent @client {
            id
          }
        }
      `;

      let likedContent = [];

      try {
        ({ likedContent } = cache.readQuery({ query }));
      } catch (e) {
        likedContent = [];
      }

      const node = {
        id: input.nodeId,
        __typename: input.nodeId.split(':')[0],
        isLiked: input.operation === 'Like',
      };

      const isLiked = likedContent.find(({ id }) => id === input.nodeId);

      if (input.operation === 'Unlike' && isLiked) {
        likedContent = likedContent.filter(({ id }) => id !== input.nodeId);
      } else if (!isLiked) {
        likedContent.push(node);
      }

      cache.writeQuery({ query, data: { likedContent } });
      return node;
    },
    mediaPlayerPlayNow: (root, trackInfo, { cache }) => {
      const query = gql`
        query {
          mediaPlayer {
            isFullscreen
          }
        }
      `;
      const mediaTrack = merge(
        {
          __typename: 'MediaPlayerTrack',
          parentId: null,
          mediaSource: null,
          posterSources: null,
          title: null,
          artist: null,
          isVideo: false,
        },
        trackInfo
      );

      const { mediaPlayer } = cache.readQuery({ query });

      const newMediaPlayerState = {
        __typename: 'MediaPlayerState',
        isPlaying: true,
        isVisible: true,
        isFullscreen: mediaTrack.isVideo
          ? true
          : (mediaPlayer && mediaPlayer.isFullscreen) || false,
        currentTrack: mediaTrack,
        currentTime: 0,
        showVideo: mediaTrack.isVideo,
        muted: false,
      };

      if (
        // if same track
        mediaPlayer &&
        mediaPlayer.currentTrack &&
        mediaPlayer.currentTrack.mediaSource.uri === mediaTrack.mediaSource.uri
      ) {
        // use the same Id
        newMediaPlayerState.currentTrack.id = mediaPlayer.currentTrack.id;
      } else {
        // otherwise reset progress and use new Id
        newMediaPlayerState.currentTrack.id = trackId;
        newMediaPlayerState.progress = null;
        trackId += 1;
      }

      cache.writeData({
        query,
        data: {
          mediaPlayer: newMediaPlayerState,
        },
      });

      track({
        eventName: events.UserPlayedMedia,
        properties: {
          uri: mediaTrack.uri,
          title: mediaTrack.title,
          type: mediaTrack.isVideo ? 'Video' : 'Audio',
        },
      });
      return null;
    },
    mediaPlayerUpdateState: (
      root,
      { isPlaying, isFullscreen, isVisible, showVideo, muted },
      { cache }
    ) => {
      const query = gql`
        query {
          mediaPlayer @client {
            isPlaying
            isFullscreen
            isVisible
            showVideo
            muted
          }
        }
      `;
      const { mediaPlayer } = cache.readQuery({ query });
      cache.writeQuery({
        query,
        data: {
          mediaPlayer: merge(mediaPlayer, {
            isPlaying,
            isFullscreen,
            isVisible,
            showVideo,
            muted,
            __typename: 'MediaPlayerState',
          }),
        },
      });
      return null;
    },
    mediaPlayerSetPlayhead: (root, { currentTime }, { cache }) => {
      const query = gql`
        query {
          mediaPlayer @client {
            currentTime
          }
        }
      `;
      const { mediaPlayer } = cache.readQuery({ query });
      cache.writeQuery({
        query,
        data: {
          mediaPlayer: {
            __typename: 'MediaPlayerState',
            currentTime:
              currentTime || get(mediaPlayer.progress, 'currentTime') || 0,
          },
        },
      });
      return null;
    },
    updateDevicePushId: async (root, { pushId }, { cache, client }) => {
      const query = gql`
        query {
          pushId @client
        }
      `;
      cache.writeQuery({
        query,
        data: {
          pushId,
        },
      });

      const { data: { isLoggedIn } = {} } = await client.query({
        query: getIsLoggedIn,
      });

      if (isLoggedIn) {
        updatePushId({ pushId });
      }
      return null;
    },

    cacheMarkLoaded: async (root, args, { cache, client }) => {
      cache.writeQuery({
        query: CACHE_LOADED,
        data: {
          cacheLoaded: true,
        },
      });
      const { data: { isLoggedIn } = {} } = await client.query({
        query: getIsLoggedIn,
      });

      const { pushId } = cache.readQuery({
        query: gql`
          query {
            pushId @client
          }
        `,
      });

      if (isLoggedIn && pushId) {
        updatePushId({ pushId });
      }
      return null;
    },
  },
};
