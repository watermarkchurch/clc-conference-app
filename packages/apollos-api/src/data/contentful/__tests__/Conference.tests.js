import { apolloServer } from 'apollos-api/src/server';
import { createTestClient } from 'apollo-server-testing';
import fetch from 'jest-fetch-mock';

describe('Conference', () => {
  beforeEach(() => {
    fetch.mockContentful();
  });

  it('gets the conference', async () => {
    const getConference = `
      query {
        conference {
          id
          title
          code
        }
      }
    `;

    const { query } = createTestClient(apolloServer);
    const result = await query({ query: getConference });
    expect(result).toMatchSnapshot();
  });

  describe('UpNext', () => {
    it('gets the first event', async () => {
      Date.now = jest.fn(() => 1546322400000);

      const getConference = `
        query {
          conference {
            upNext {
              id
              title
              ...on Event {
                startTime
                endTime
              }
            }
          }
        }
      `;

      const { query } = createTestClient(apolloServer);
      const result = await query({ query: getConference });
      expect(result).toMatchSnapshot();
    });

    it('gets the second event', async () => {
      Date.now = jest.fn(() => 1554139800000);

      const getConference = `
        query {
          conference {
            upNext {
              id
              title
              ...on Event {
                startTime
                endTime
              }
            }
          }
        }
      `;

      const { query } = createTestClient(apolloServer);
      const result = await query({ query: getConference });
      expect(result).toMatchSnapshot();
    });

    it('gets nested liked events', async () => {
      Date.now = jest.fn(() => 1554307200000);

      const getConference = `
        query {
          conference {
            upNext(likedIds: ["Event:0cadc8706009779ee2d54b650930695e7669398b5e6ad4d34a375a728a696382"]) {
              id
              title
              ...on Event {
                startTime
                endTime
              }
            }
          }
        }
      `;

      const { query } = createTestClient(apolloServer);
      const result = await query({ query: getConference });
      expect(result).toMatchSnapshot();
    });
  });
});
