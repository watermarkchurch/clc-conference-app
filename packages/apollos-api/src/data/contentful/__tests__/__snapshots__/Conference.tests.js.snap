// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conference UpNext gets nested liked events 1`] = `
Object {
  "data": Object {
    "conference": Object {
      "upNext": Object {
        "endTime": "2019-04-03T00:00-05:00",
        "id": "Event:0cadc8706009779ee2d54b650930695e7669398b5e6ad4d34a375a728a696382",
        "startTime": "2019-04-03T11:00-05:00",
        "title": "Volunteers: Proper Care and Feeding (and Finding!)",
      },
    },
  },
  "errors": undefined,
  "extensions": Object {
    "cacheControl": Object {
      "hints": Array [
        Object {
          "maxAge": 3600,
          "path": Array [
            "conference",
          ],
          "scope": undefined,
        },
        Object {
          "maxAge": 300,
          "path": Array [
            "conference",
            "upNext",
          ],
          "scope": undefined,
        },
      ],
      "version": 1,
    },
  },
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {
        "Cache-Control": Array [
          "max-age=300, public",
        ],
      },
    },
  },
}
`;

exports[`Conference UpNext gets the first event 1`] = `
Object {
  "data": Object {
    "conference": Object {
      "upNext": Object {
        "endTime": "2019-04-01T12:30-05:00",
        "id": "Event:b2a8ecde85596a465387ee155335c69451ff6f0cabd4e4eb9b6f5e7239cf3eb0",
        "startTime": "2019-04-01T11:00-05:00",
        "title": "PRE-CONFERENCE WORKSHOP REGISTRATION OPENS",
      },
    },
  },
  "errors": undefined,
  "extensions": Object {
    "cacheControl": Object {
      "hints": Array [
        Object {
          "maxAge": 3600,
          "path": Array [
            "conference",
          ],
          "scope": undefined,
        },
        Object {
          "maxAge": 300,
          "path": Array [
            "conference",
            "upNext",
          ],
          "scope": undefined,
        },
      ],
      "version": 1,
    },
  },
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {
        "Cache-Control": Array [
          "max-age=300, public",
        ],
      },
    },
  },
}
`;

exports[`Conference UpNext gets the second event 1`] = `
Object {
  "data": Object {
    "conference": Object {
      "upNext": Object {
        "endTime": "2019-04-01T13:00-05:00",
        "id": "Event:4c9a0152c5d5bc3ed8eacb1356e954c005fae3f2f1041d319ccea92926e2f26f",
        "startTime": "2019-04-01T12:30-05:00",
        "title": "Workshop Kickoff",
      },
    },
  },
  "errors": undefined,
  "extensions": Object {
    "cacheControl": Object {
      "hints": Array [
        Object {
          "maxAge": 3600,
          "path": Array [
            "conference",
          ],
          "scope": undefined,
        },
        Object {
          "maxAge": 300,
          "path": Array [
            "conference",
            "upNext",
          ],
          "scope": undefined,
        },
      ],
      "version": 1,
    },
  },
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {
        "Cache-Control": Array [
          "max-age=300, public",
        ],
      },
    },
  },
}
`;

exports[`Conference gets the conference 1`] = `
Object {
  "data": Object {
    "conference": Object {
      "code": "CLC2019",
      "id": "Conference:ddb7ea3e5e7bb7b45de5ceaa99de49ce8dfcf17f1f6a4bcaca552551b7633462",
      "title": "Church Leaders Conference 2019",
    },
  },
  "errors": undefined,
  "extensions": Object {
    "cacheControl": Object {
      "hints": Array [
        Object {
          "maxAge": 3600,
          "path": Array [
            "conference",
          ],
          "scope": undefined,
        },
        Object {
          "maxAge": 3600,
          "path": Array [
            "conference",
            "id",
          ],
          "scope": undefined,
        },
        Object {
          "maxAge": 3600,
          "path": Array [
            "conference",
            "title",
          ],
          "scope": undefined,
        },
        Object {
          "maxAge": 3600,
          "path": Array [
            "conference",
            "code",
          ],
          "scope": undefined,
        },
      ],
      "version": 1,
    },
  },
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {
        "Cache-Control": Array [
          "max-age=3600, public",
        ],
      },
    },
  },
}
`;
